wolfssl/wolfcrypt/port/arm/cryptoCell.h

```diff
@@ -0,0 +1,122 @@
+/* cryptoCell.h
+ *
+ * Copyright (C) 2019 wolfSSL Inc.
+ *
+ * This file is part of wolfSSL.
+ *
+ * wolfSSL is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * wolfSSL is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1335, USA
+ */
+
+#ifndef WOLFSSL_CRYPTOCELL_H
+#define WOLFSSL_CRYPTOCELL_H
+
+#if defined(WOLFSSL_CRYPTOCELL)
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <wolfssl/wolfcrypt/types.h>
+#include "nrf.h"
+#include "sns_silib.h"
+
+
+#ifndef NO_SHA256
+/* for CRYS_HASH_* interfaces */
+#include "crys_hash.h"
+#include "crys_hash_error.h"
+
+#endif
+
+#ifndef NO_AES
+#include "ssi_aes.h"
+
+typedef struct aes_context_t {
+    SaSiAesUserContext_t user_ctx;    /* CC310 AES User context */
+    SaSiAesUserKeyData_t key;         /* CC310 AES key structure */
+    SaSiAesEncryptMode_t mode;        /* encrypt or decrypt */
+} aes_context_t;
+#define CC310_MAX_LENGTH_DMA_AES    (0xFFF0)
+#endif /* NO_AES */
+
+#if !defined(WC_NO_RNG)
+#if defined(WOLFSSL_nRF5x_SDK_15_2)
```

Comment:
Why use such a specific macro guard? `WOLFSSL_nRF5x_SDK_15_2`. Why not `WOLFSSL_NRF5x_SDK`?